conf/Config.js

```diff
@@ -166,25 +166,34 @@ class Config extends EventEmitter {
         return this.bootstrapList;
     }
 
-    setLocations(locationConstraints) {
+    setIngestionBuckets(locationConstraints, buckets) {
         // NOTE: currently only s3 connector (s3c 7.4.2 and above) are supported
         const ingestionTypeMatch = Object.assign({}, locationTypeMatch, {
             'location-scality-ring-s3-v1': 'scality_s3',
         });
 
         // refresh/reset
-        this.locations = {};
-        Object.keys(locationConstraints).forEach(key => {
-            const locItem = locationConstraints[key];
-            this.locations[key] = Object.assign({}, locItem.details, {
-                locationType: ingestionTypeMatch[locItem.locationType],
-            });
+        this.ingestionBuckets = [];
+        buckets.forEach(bucket => {
+            const { name, ingestion, locationConstraint } = bucket;
+            const locationInfo = locationConstraints[locationConstraint];
```

Comment:
The buckets should be buckets created from a given zenko storage location so I don't think this will be a problem, but i'm not completely sure, so will add a check with a debug log and will skip the bucket. Thanks!